### PR TITLE
 Gh-2924 Fix send null payload with KafkaTemplate#send(Message), and …

### PR DESCRIPTION
**Pull request by pswrdf**
…JsonMessageConverter

fixes: https://github.com/spring-projects/spring-kafka/issues/2924

### CODE REVIEW

**Comment by sobychacko** [comment]
@pswrdf Thanks for the PR! Can you add a unit test to verify the changes? Or point to an existing test where this code is exercised.

**Review by artembilan — CHANGES_REQUESTED** [review]
Hi @pswrdf !

Thank you for contribution.
Please, make yourself familiar with commit message rules we try to follow: https://cbea.ms/git-commit/.

I don't say that you need to change anything right now if you don't want, but just for the future to avoid comments from me like this 😉 

@sobychacko ,

This fix is supposed to be back-ported to `3.0.x`.
When you merge this, please, consider to include `Fixes: gh-2924` like into commit message.
This way Backport Bot GHA workflow will create a respective back-port issue for for current `3.0.x` milestone. However don't merge yet, please, I'm working in that branch right now to migrate it into GHA.

**Review by artembilan — CHANGES_REQUESTED** [review]
So, now CI/CD infra is set up for `3.0.x`.
Therefore we are good to cherry-pick after merging this.

1. Add some unit test, please. See `BatchListenerConversionTests` where we use one of those `JsonMessageConverter` impls.
2. Add your name to the `@author` list of all the affected classes.
3. Add also `Fixes: gh-2924` into next commit message as I asked before.
4. Add `**Cherry-pick to `3.0.x`**` into the next commit message.

Thanks

File: spring-kafka/src/main/java/org/springframework/kafka/support/converter/JsonMessageConverter.java
Comment (sobychacko) [inline spring-kafka/src/main/java/org/springframework/kafka/support/converter/JsonMessageConverter.java:1]: Missing `author` tag in this class.

Comment (artembilan) [reply spring-kafka/src/main/java/org/springframework/kafka/support/converter/JsonMessageConverter.java:1]: There is no need. 
This class has lost its changes.

Comment (pswrdf) [reply spring-kafka/src/main/java/org/springframework/kafka/support/converter/JsonMessageConverter.java:1]: it was reverted in the second commit.

I think it's done.

**Comment by pswrdf** [comment]
I think it's done.

Would I squash commits? It needs to change email also.

**Comment by artembilan** [comment]
... and cherry-picked to `3.0.x`

**Pull request merged**
